test/test_io.py

```diff
@@ -43,6 +72,27 @@ def test_create_destroy(self):
         del bootstrap
         self.assertTrue(bootstrap_shutdown_event.wait(TIMEOUT))
 
+    def test_create_destroy_singleton(self):
+        bootstrap = ClientBootstrap.get_or_create_static_default()
+
+        # ensure shutdown_event fires
+        bootstrap_shutdown_event = bootstrap.shutdown_event
+        del bootstrap
+
+        ClientBootstrap.release_static_default()
+        EventLoopGroup.release_static_default()
+        DefaultHostResolver.release_static_default()
+
+        self.assertTrue(bootstrap_shutdown_event.wait(TIMEOUT))
+
+    def test_init_singleton_is_singleton(self):
+        client_one = ClientBootstrap.get_or_create_static_default()
+        client_two = ClientBootstrap.get_or_create_static_default()
+        self.assertTrue(client_one == client_two)
+        ClientBootstrap.release_static_default()
+        EventLoopGroup.release_static_default()
+        DefaultHostResolver.release_static_default()
```

Comment:
Add these 3 `release_static_default()` calls to this function:
[awscrt._test.check_for_leaks()](https://github.com/awslabs/aws-crt-python/blob/cc651452fa1900c140fedde0be65962b5b897396/awscrt/_test.py#L88)

rambling background about cleanup in aws-crt-python...

aws-crt-python is different than the other language bindings in that it doesn't try too hard to clean things up on exit, so there's no pre-existing "ok clean it all up" function for the library. But we do have this private function we call at the end of every test that checks for memory leaks, so let's put it there for now.

Some background on why we don't even try to clean up...
We used to! But Python developers like to write simple applications that exit the moment they get what they want (or exit immediately because of error), and we kept getting crashes when the library tried to clean things up on exit but the Python application was still running stuff on another thread. So now we try and tolerate ["sloppy shutdown"](https://github.com/awslabs/aws-crt-python/pull/116)